### DISCUSSION TITLE
Mapping cmd+number to tmux

### DISCUSSION THREAD

**Discussion by pwntester** [ghostty-org/ghostty#3447] [Q&A] [answered]
Im trying to map cmd + a number to the tmux prefix + the same number with:

```
keybind = cmd+1=text:\x021
keybind = cmd+2=text:\x022
keybind = cmd+3=text:\x023
keybind = cmd+4=text:\x024
keybind = cmd+5=text:\x025
keybind = cmd+6=text:\x026
keybind = cmd+7=text:\x027
keybind = cmd+8=text:\x028
keybind = cmd+9=text:\x029
```

These mappings are not working, however replacing `cmd` with `opt` works so it seems the action is correct and perhaps cmd+number is reserved somehow.

The same mappings are working correctly in wezterm so my guess is that ghostty is using these mappings on its own. Any feedback is appreciated!

**Comment by mitchellh** [comment, marked as answer]
`cmd+one`, `cmd+two`. This is awkward so we'll add aliases for the numbers, but noting this should work.

**Reply by pwntester** [reply]
That worked, thanks!

**Reply by gonstoll** [reply]
Side question (sorry to interrupt): I've been pulling my hair off trying to understand how to map those escape sequences to actual keys. Any secret you care to share for knowing what a key combination means in escape sequence? Like, how do I know that `Ctrl-f` is equivalent to `\x06`? So far, the only thing I've found that actually helped was asking chatgpt, and it was not easy to get it to tell me what I wanted :/

**Reply by m00p1ng** [reply]
> Side question (sorry to interrupt): I've been pulling my hair off trying to understand how to map those escape sequences to actual keys. Any secret you care to share for knowing what a key combination means in escape sequence? Like, how do I know that `Ctrl-f` is equivalent to `\x06`? So far, the only thing I've found that actually helped was asking chatgpt, and it was not easy to get it to tell me what I wanted :/

You can see the complete list of `ASCII Device Control Codes` [here](https://www3.rocketsoftware.com/bluezone/help/v42/en/bzadmin/APPENDIX_C/ASCII_Character_Set.htm)

**Comment by mitchellh** [comment]
#4146 btw

**Comment by JuaniRaggio** [comment]
This didn't work for me, I even tried unbinding default mappings using `keybind = cmd+one=unbind` and still didn't work. The way I fixed it was first unbinding default mapping, but for some reason you have to use `digit_1`, `digit_2`, etc. to unbind them, this is what I mean and worked for me:

```sh
keybind = cmd+digit_1=unbind
keybind = cmd+digit_2=unbind
keybind = cmd+digit_3=unbind
keybind = cmd+digit_4=unbind
keybind = cmd+digit_5=unbind
keybind = cmd+digit_6=unbind
keybind = cmd+digit_7=unbind
keybind = cmd+digit_8=unbind
keybind = cmd+digit_9=unbind

# Expecting you to have \x02 (ctl+b) as tmux prefix
keybind = cmd+one=text:\x02\x31
keybind = cmd+two=text:\x02\x32
keybind = cmd+three=text:\x02\x33
keybind = cmd+four=text:\x02\x34
keybind = cmd+five=text:\x02\x35
keybind = cmd+six=text:\x02\x36
keybind = cmd+seven=text:\x02\x37
keybind = cmd+eight=text:\x02\x38
keybind = cmd+nine=text:\x02\x39
```

**Reply by rhodes-b** [reply]
Because the original post is 11 months old and 1.2 introduced a significant keybind rework to use w3c keycode names
